homework/App.js

```diff
@@ -12,18 +12,27 @@ class App {
    * @param {string} url The GitHub URL for obtaining the organization's repositories.
    */
   async initialize(url) {
-    // Add code here to initialize your app
     // 1. Create the fixed HTML elements of your page
     // 2. Make an initial XMLHttpRequest using Util.fetchJSON() to populate your <select> element
-
     const root = document.getElementById('root');
-
-    Util.createAndAppend('h1', root, { text: 'It works!' }); // TODO: replace with your own code
-
+    const header = Util.createAndAppend('header', root, { class: 'header' });
+    Util.createAndAppend('p', header, { text: 'HYF Repositories' });
+    const select = Util.createAndAppend('select', header, { id: 'select' });
+    Util.createAndAppend('div', root, { id: 'repo-container' });
+    select.addEventListener('change', event => {
+      this.fetchContributorsAndRender(event.target.value);
+    });
     try {
       const repos = await Util.fetchJSON(url);
-      this.repos = repos.map(repo => new Repository(repo));
-      // TODO: add your own code here
+      this.repos = repos
+        .sort((a, b) => a.name.localeCompare(b.name))
+        .map(repo => new Repository(repo));
+
+      repos.forEach((repository, index) => {
+        Util.createAndAppend('option', select, { text: repository.name, value: index });
+      });
+      const index = select.value;
+      this.fetchContributorsAndRender(index);
```

Comment:
Why not call it like this:

```js
this.fetchContributorsAndRender(select.value);
```

and get rid of the `index` variable that you do not use elsewhere.

Reply:
Thanks for your feedback! I just want to make it clear for me "index = select.value"